packages/wxt/src/virtual/unlisted-script-entrypoint.ts

```diff
@@ -2,17 +2,37 @@ import definition from 'virtual:user-unlisted-script-entrypoint';
 import { logger } from '../utils/internal/logger';
 import { initPlugins } from 'virtual:wxt-plugins';
 
-const result = (async () => {
+const result = (() => {
   try {
     initPlugins();
-    return await definition.main();
+  } catch (err) {
+    logger.error(
+      `Failed to initialize plugins for "${import.meta.env.ENTRYPOINT}"`,
+      err,
+    );
+    throw err;
+  }
+  let result;
+  try {
+    result = definition.main();
+
+    if (result && typeof result === 'object' && 'then' in result) {
```

Comment:
I've seen other devs check for promises like this, but never understand why. Why not just check `result instanceof Promise`?

```suggestion
    if (result instanceof Promise) {
```